_docs/technical/2017-01-04-updater.md

```diff
@@ -123,41 +147,69 @@ for more information.
 ## Update Proposal Approval
 
 A very important part of implementation of the update mechanism is
-the part that works with genesis blocks for epochs and applies updates.
+the part that works with genesis blocks for epochs.
 This logic resides
-[in this well-documented function](https://github.com/input-output-hk/cardano-sl/blob/22360aa45e5dd82d0c87872d8530217fc3d08f4a/src/Pos/Update/Poll/Logic/Softfork.hs#L67).
+[in this well-documented function](https://github.com/input-output-hk/cardano-sl/blob/6b5eda44e5942599a9781e5ad3f51eb820665b83/src/Pos/Update/Poll/Logic/Softfork.hs#L68).
 We explain terminology related to this process below.
 
 ### Threshold
 
 `softforkResolutionThreshold` is a threshold (referred to as «threshold»
-in the code) which, for an update proposal with version `v`, says ???
+in the code), which allows to define stable stake for each stakeholder
+as their stake `k` blocks ago. See more detailed description in
+[research overview](cardano/update-mechanism/#soft-fork-updates).
+
+
+### Proposal states
+
+**Software** version (which can be found in proposals in blocks)
+can be in one of the states described bellow.
+
+#### Undecided
+
+It means that update proposal with given **software** version is contained in
```

Comment:
Better remove _with given software version_ part (see my comments to PR itself).